### PR TITLE
[model, ci, agent] feat: migrate qwen3_omni_moe to transformers v5

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

> Migrates `qwen3_omni_moe` (thinker-only; talker training still out of scope) to the transformers v5 patchgen path (Pattern B: v4↔v5 coexist, MoE) at the 5.2.0 gate. GPU-only — no NPU sibling for this model yet. Builds on #666 (qwen3_vl_moe v5) and reuses the same MoE + checkpoint-converter conventions.

### Checklist Before Starting

- Search for relative PRs/issues and link here: builds on #666 (qwen3_vl_moe v5 migration)
- PR title follows `[{modules}] {type}: {description}` format

### Test

> - `tests/models/test_models_patch.py`: added `qwen3_omni_moe` to `_TEST_CASES_TRANSFORMERS_V5`.
> - `tests/distributed/test_dummy_forward.py`: added `qwen3_omni_moe_v5` to `_omni_cases`.
> - `tests/e2e/test_e2e_parallel.py`: added `qwen3_omni_moe_v5` to `qwen3omni_test_cases`.
> - `tests/models/test_checkpoint_tensor_converter.py`: new `TestQwen3OmniMoeConverter*` block covering `can_handle` across thinker/talker/standalone prefixes, HF→v5 transpose, v5 passthrough, shape/rank guards, and the factory for all three config shapes.

### API and Usage Example

> N/A — internal migration. Existing `qwen3_omni_moe` usage continues to work; v4 path (`modeling_qwen3_omni_moe.py` + legacy `parallel_plan.py` keys) is retained for transformers 4.57.3 back-compat. v5 path is selected automatically when `transformers >= 5.2.0`.

### Design & Code Changes

> - **Patchgen config**: `qwen3_omni_moe_gpu_patch_gen_config.py` with 16 patches; generated output at `generated/patched_modeling_qwen3_omni_moe_gpu.{py,diff}` (do not edit by hand).
>   - Vision encoder: `lru_cached` rot_pos_ids, tensorized `fast_pos_embed_interpolate`, attention + block `max_seqlen`, `dummy_forward`.
>   - Audio encoder: SP-friendly Conformer-style forward + `dummy_forward` on `_dummy_input_features`.
>   - Text decoder: async Ulysses attention, deepstack fusion kept out-of-place under SP scatter.
>   - Thinker `ForConditionalGeneration.forward`: SP-aware multimodal scatter, fused CE via `self.loss_function`, MoE aux_loss.
>   - Top-level `Qwen3OmniMoeForConditionalGeneration.forward`: thin wrapper returning the thinker output directly for training.
> - **MoE expert layout**: `Qwen3OmniMoeThinkerTextExperts` replaced with fused layout (`gate_up_proj` `[E, 2*I, H]`, `down_proj` `[E, H, I]`) dispatching `fused_moe_forward` on `config._moe_implementation`. The `@use_experts_implementation` decorator is dropped so veomni's fused kernel path is reached.
> - **`_moe_implementation` propagation**: override `Qwen3OmniMoeThinkerModel.__init__` to propagate top-level → `config.thinker_config` → `config.text_config` before `super().__init__` so expert modules see it.
> - **`get_rope_index` patch**: compute per-video `use_audio_in_video` via `audio_seqlens` (Qwen2.5-Omni convention, preserved from v4) and tolerate `attention_mask=None` at the data boundary.
> - **Checkpoint tensor converter** (`checkpoint_tensor_converter.py`): HF ships thinker/talker experts under the same fused key names as v5 but in transposed layout (`[E, H, 2*I]` / `[E, I, H]`). Dim-1 shape dispatch transposes HF, passes v5-native through, hard-errors on unrecognized shapes.
> - **`__init__.py` v5 branch**: imports generated classes and attaches `_create_checkpoint_tensor_converter` as `staticmethod` on `Qwen3OmniMoeForConditionalGeneration`, `Qwen3OmniMoeThinkerForConditionalGeneration`, and `Qwen3OmniMoeThinkerTextModel` (all three are loadable standalone). Also fixes an upstream HF bug where `Qwen3OmniMoePreTrainedModel._no_split_modules` references a nonexistent `Qwen3OmniMoeDecoderLayer` by overriding on the wrapper with the real class names.
> - **`parallel_plan.py`**: EP `Shard(0)` keys updated from split `gate_proj`/`up_proj` to fused `gate_up_proj`, matching the new expert layout.
> - **`veomni/patchgen/codegen.py`**: when `exclude_from_output` drops classes, filter them out of module-level `__all__ = [...]` so the generated module doesn't claim to export missing names (avoids F822 and `from ... import *` breakage).
> - **`.agents/skills/veomni-migrate-transformers-v5/SKILL.md`**: small doc update capturing MoE/omni conventions encountered during this migration.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- [x] Applied pre-commit checks
- [x] Added/updated documentation
- [x] Added tests to CI workflow (or explained why not feasible)

### CODE REVIEW

**Comment by TimYangst** [comment]
@gemini-code-assist please take another review.

File: veomni/models/transformers/qwen3_omni_moe/qwen3_omni_moe_gpu_patch_gen_config.py [hunk 1]
```diff
@@ -0,0 +1,1460 @@
...
+    aux_loss = None
+    if output_router_logits:
+        aux_loss = load_balancing_loss_func(
+            outputs.router_logits,
+            self.num_experts,
+            self.num_experts_per_tok,
+            attention_mask,
+        )
+        if labels is not None:
+            loss += self.router_aux_loss_coef * aux_loss.to(loss.device)
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply veomni/models/transformers/qwen3_omni_moe/qwen3_omni_moe_gpu_patch_gen_config.py]: updated.

File: veomni/models/transformers/qwen3_omni_moe/qwen3_omni_moe_gpu_patch_gen_config.py [hunk 2]
```diff
@@ -0,0 +1,1460 @@
...
+
+    return Qwen3OmniMoeThinkerCausalLMOutputWithPast(
+        loss=loss,
+        logits=logits,
+        aux_loss=aux_loss,
+        hidden_states=outputs.hidden_states,
+        attentions=outputs.attentions,
+        past_key_values=outputs.past_key_values,
+        rope_deltas=self.rope_deltas,
+    )
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply veomni/models/transformers/qwen3_omni_moe/qwen3_omni_moe_gpu_patch_gen_config.py:1353]: Fixed

**Review by piyifan123 — APPROVED** [review]
